### DISCUSSION TITLE
Lesson 4: FundMe Project

### DISCUSSION THREAD

**Discussion by spomaria** [Cyfrin/foundry-full-course-cu#1524] [General]
After compiling the FundeMe.sol contract without errors, anytime the fund function is called, I get this error:
```
transact to FundMe.fund errored: Error occurred: revert.

revert
	The transaction has been reverted to the initial state.
Note: The called function should be payable if you send value and the value you send should be less than your current balance.
Debug the transaction to get more information.
```
This is the line of code that should get the price feed from Chainlink
```solidity
function getPrice() internal view returns (uint256) {
        // Get the Address and ABI of the Contract the stores the price of ETH
        // on ChainLink website. From Sepolia TestNet we have
        // Address 0x694AA1769357215DE4FAC081bf1f309aDC325306 
        // ABI 
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        // The 'latestRoundData' method returns five values. 
        // However, we are only interested in 'price'
        (, int256 price, , , ) = priceFeed.latestRoundData();
        // price of ETH in terms of USD
        // We employ a technique call type-casting to change a variable type
        // from int256 to uint256
        return uint256(price * 1e10);
    }
```

I have the thought that the Contract is not getting the price feed from Chainlink as supposed because if I edit the contract to not check the price feed from Chainlink, the fund function works find. I will be grateful to have suggestions on how to make my contract get the price feed from Chainlink. Thanks

**Comment by EngrPips** [comment]
Hello @spomaria , do you encounter this error when you call the `fund` function on the `anvil` chain or `sepolia` test net ?

**Reply by spomaria** [reply]
Hi @EngrPips I get the error on `sepolia` test net.

**Reply by EngrPips** [reply]
Have you confirmed that you have enough sepolia ether to cover the payment when you call the `fund` function ?

**Reply by spomaria** [reply]
I have about 0.49 Sepolia ETH, I believe that should be enough

**Reply by EngrPips** [reply]
Can you maybe push your code to github, so i can clone, set it up and try debug this issues on my machine

**Reply by spomaria** [reply]
Kindly find the link to the github repo 

`https://github.com/spomaria/FundMe-Exercise`

Kindly note that code was slightly adjusted to bypass getting price feed from chainlink by commenting out the line of code thus

```solidity
function fund() public payable {
        // allow users to send $
        // set a minimum amount a user can send
        // 1. How do we send ETH to this contract?
        //  We make the function payable so that is can send ETH
        // require(msg.value.getConversionRate() > MINIMUM_USD, "didn't send enough ETH"); // 1e18 = 1ETH
        require(msg.value > MINIMUM_USD, "didn't send enough ETH"); // 1e18 = 1ETH
        // Include this address in the list of funders
        funders.push(msg.sender);
        addressToAmountFunded[msg.sender] += msg.value;
    }
```

**Reply by EngrPips** [reply]
Alright, thanks for sharing. i would for sure get back to you with whatever i find
